In [90]:
# import libraries
import requests
from bs4 import BeautifulSoup as bs

In [91]:
url = 'https://www.ryanscomputers.com/category/laptop-all-laptop?limit=2000&osp=0&st=0'

# query the website and return the html to the variable ‘response’
response = requests.get(url).text

soup = bs(response, "html.parser")

In [92]:
print(soup)

IOPub data rate exceeded.
The notebook server will temporarily stop sending output
to the client in order to avoid crashing it.
To change this limit, set the config variable
`--NotebookApp.iopub_data_rate_limit`.

Current values:
NotebookApp.iopub_data_rate_limit=1000000.0 (bytes/sec)
NotebookApp.rate_limit_window=3.0 (secs)



In [93]:
product_names = []
product_ids = []
processor_types = []
product_generations = []
product_rams = []
product_storages = []
product_graphics_x = []
product_displays = []
product_prices = []

In [94]:
# Find all product details
for product in soup.find_all('div', class_='card-body text-center'):
    # Get product name, link, and ID
    product_name_link = product.find('a', {'data-bs-toggle': 'tooltip'})
    if product_name_link:
        product_info = product_name_link.get('title').split('<br>')
        product_name = product_info[0]
        product_id = product_info[1].replace('Product Id: ', '')
        product_names.append(product_name)
        product_ids.append(product_id)
        print(f'Product Name: {product_name}')
        print(f'Product ID: {product_id}')

    # Get product attributes
    product_attributes = product.find_all('li', class_='card-text p-0 m-0 fw-normal')
    for attribute in product_attributes:
        attribute_text = attribute.text
        if 'Processor Type' in attribute_text:
            processor_type = attribute_text.replace("Processor Type. - ", "")
            processor_types.append(processor_type)
            print(f'Processor Type: {attribute_text.replace("Processor Type. - ", "")}')
        elif 'Generation' in attribute_text:
            product_generation = attribute_text.replace("Generation - ", "")
            product_generations.append(product_generation)
            print(f'Generation: {attribute_text.replace("Generation - ", "")}')
        elif 'RAM' in attribute_text:
            product_ram = attribute_text.replace("RAM - ", "")
            product_rams.append(product_ram)
            print(f'RAM: {attribute_text.replace("RAM - ", "")}')
        elif 'Storage' in attribute_text:
            product_storage = attribute_text.replace("Storage - ", "")
            product_storages.append(product_storage)
            print(f'Storage: {attribute_text.replace("Storage - ", "")}')
        elif 'Graphics Memory' in attribute_text:
            product_graphics = attribute_text.replace("Graphics Memory - ", "")
            product_graphics_x.append(product_graphics)
            print(f'Graphics Memory: {attribute_text.replace("Graphics Memory - ", "")}')
        elif 'Display Size (Inch)' in attribute_text:
            product_display = attribute_text.replace("Display Size (Inch) - ", "")
            product_displays.append(product_display)
            print(f'Display Size (Inch): {attribute_text.replace("Display Size (Inch) - ", "")}')

    # Get product price
    product_price = product.find('p', class_='pr-text cat-sp-text pb-1')
    if product_price:
        price = product_price.text.replace('Tk ', '')
        product_prices.append(price)
        print(f'Product Price: {price}')

Streaming output truncated to the last 5000 lines.
Product Price: 0 
Product Name: MSI Vector GP77 13VG Intel Core i7 13700H 16GB RAM 2TB SSD 17.3 Inch 2K QHD Display Core Black Gaming Laptop
Product ID:  33.02.033.1235
Processor Type: Core i7
Generation: 13th (Intel)
RAM: 16GB
Storage: 2TB SSD
Graphics Memory: 8GB
Display Size (Inch): 17.3
Product Price: 0 
Product Name: MSI Katana 17 B13VEK Intel Core i7 13620H 16GB RAM 1TB SSD 17.3 Inch FHD Display Black Gaming Laptop
Product ID:  33.02.033.1236
Processor Type: Core i7
Generation: 13th (Intel)
RAM: 16GB
Storage: 1TB SSD
Graphics Memory: 6GB
Display Size (Inch): 17.3
Product Price: 0 
Product Name: MSI Sword 15 A12VF Intel Core i7 12650H 16GB RAM 1TB SSD 15.6 Inch FHD Display White Gaming Laptop
Product ID:  33.02.033.1238
Processor Type: Core i7
Generation: 12th (Intel)
RAM: 16GB
Storage: 1TB SSD
Graphics Memory: 8GB
Display Size (Inch): 15.6
Product Price: 0 
Product Name: Acer Aspire 3 A315-23 AMD Ryzen 3 3250U 8GB RAM 256GB SSD 1

In [95]:
product_names[:5]

['Lenovo IdeaPad D330 10IGL Intel CDC N4020 4GB RAM 128GB eMMC 10.1 Inch HD IPS Touch Display Mineral Grey Laptop',
 'HP 15s-eq1578AU AMD Athlon Silver 3050U 8GB  256GB SSD 15.6 Inch FHD Display Silver Laptop',
 'Acer Aspire 3 A315-510P-38RH (13th Gen Standard) Intel Core i3 N305 8GB RAM, 512GB SSD 15.6 Inch FHD Display Pure Silver Laptop',
 'Lenovo IdeaPad 1 15ADA7 AMD Ryzen 3 3250U 8GB RAM 512GB SSD 15.6 Inch FHD Display Cloud Grey Laptop',
 'Lenovo IdeaPad Slim 3i 15ITL 11th Gen Intel Core i3 1115G4 4GB RAM 256GB SSD 15.6 Inch FHD Display Arctic Grey Laptop']

In [96]:
product_ids[:5]

[' 33.01.200.965',
 ' 33.01.020.1490',
 ' 33.01.002.1029',
 ' 33.01.200.1009',
 ' 33.01.200.1036']

In [97]:
processor_types[:5]

['Intel Celeron', 'AMD Athlon', 'Core i3', 'Ryzen 3', 'Core i3']

In [98]:
product_generations[:5]

['Not Applicable',
 'Not Applicable',
 '13th Standard (Intel)',
 'Not Applicable',
 '11th (Intel)']

In [99]:
product_rams[:5]

['4GB', '8GB', '8GB', '8GB', '4GB']

In [100]:
product_storages[:5]

['128GB eMMC', '256GB SSD', '512GB SSD', '512GB SSD', '256GB SSD']

In [101]:
product_graphics_x[:5]

['Shared', 'Shared', 'Shared', 'Shared', 'Shared']

In [102]:
product_displays[:5]

['10.1', '15.6', '15.6', '15.6', '15.6']

In [103]:
product_prices[:5]

['34,500 ', '46,500 48,900 ', '49,800 53,000 ', '49,900 ', '51,400 ']

In [104]:
product_prices_len = len(product_prices)
product_prices_len

1786

# Export to Excel CSV

In [105]:
import csv

In [106]:
# Prepare data for CSV
rows = zip(product_names, product_ids, processor_types, product_generations, product_rams, product_storages, product_graphics_x, product_displays, product_prices)

# Write to CSV
with open('LaptopPrices.csv', 'w', newline='') as f:
    writer = csv.writer(f)
    writer.writerow(["Product Name", "Product ID", "Processor Type", "Generation", "RAM", "Storage", "Graphics", "Display", "Price"])  # writing headers
    for row in rows:
        writer.writerow(row)

In [107]:
from google.colab import files

# Download the file to your computer
files.download('LaptopPrices.csv')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [108]:
import pandas as pd
LaptopPrices = pd.read_csv("LaptopPrices.csv")

In [109]:
LaptopPrices.head()

Product Name       Product ID  \
0  Lenovo IdeaPad D330 10IGL Intel CDC N4020 4GB ...    33.01.200.965   
1  HP 15s-eq1578AU AMD Athlon Silver 3050U 8GB  2...   33.01.020.1490   
2  Acer Aspire 3 A315-510P-38RH (13th Gen Standar...   33.01.002.1029   
3  Lenovo IdeaPad 1 15ADA7 AMD Ryzen 3 3250U 8GB ...   33.01.200.1009   
4  Lenovo IdeaPad Slim 3i 15ITL 11th Gen Intel Co...   33.01.200.1036   

  Processor Type             Generation  RAM     Storage Graphics  Display  \
0  Intel Celeron         Not Applicable  4GB  128GB eMMC   Shared     10.1   
1     AMD Athlon         Not Applicable  8GB   256GB SSD   Shared     15.6   
2        Core i3  13th Standard (Intel)  8GB   512GB SSD   Shared     15.6   
3        Ryzen 3         Not Applicable  8GB   512GB SSD   Shared     15.6   
4        Core i3           11th (Intel)  4GB   256GB SSD   Shared     15.6   

            Price  
0         34,500   
1  46,500 48,900   
2  49,800 53,000   
3         49,900   
4         51,400

In [110]:
LaptopPrices.shape

(1786, 9)

**As we can see, a dataset consisting 1786 rows and 9 columns of different laptop attribues has successfully been created.**